### PR TITLE
Fixes bit, varbit, binary and datetime types

### PR DESCRIPTION
**Pull request by alishakawaguchi**
- updates CLI tests to validate destination table column values
- adds new neosync types bit, binary and datetime
- updates connection data stream endpoint to use gob
- add dynamodb cli sync integration test
- fixes dynamodb cli sync
- consolidates cli and worker connection type values

### CODE REVIEW

File: charts/neosync/README.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ A Helm chart for Neosync that contains the api, app, and worker
 | api.updateStrategy | string | `nil` | The strategy to use when rolling out new replicas |
 | api.volumeMounts | list | `[]` | Volumes that will be mounted to the deployment |
 | api.volumes | list | `[]` | Volumes that will be attached to the deployment |
-| app.analytics.enabled | bool | `true` | Enables analytics such as Posthog/Koala (if keys have been provided for them) |
+| app.analytics.enabled | bool | `true` | Enables analytics such as Posthog/Unify (if keys have been provided for them) |
```
Comment (alishakawaguchi) [inline charts/neosync/README.md:115]: I didn't add any of the unify stuff. This was on main where I originally made this branch

File: internal/neosync-types/array_test.go [hunk 2]
```diff
@@ -98,21 +101,13 @@ func Test_NeosyncArray(t *testing.T) {
 		value, err := array.ValuePgx()
 		require.NoError(t, err)
 
-		values, ok := value.([]interface{})
+		fmt.Println("value", value, reflect.TypeOf(value))
```
Comment (nickzelei) [inline internal/neosync-types/array_test.go:104]: t.log?

File: worker/pkg/benthos/sql/processor_neosync_mssql.go [hunk 3]
```diff
@@ -137,3 +145,14 @@ func getMssqlValue(value any) (any, error) {
 
 	return value, nil
 }
+
+func getMssqlNeosyncValue(root any) (value any, isNeosyncValue bool, err error) {
+	if valuer, ok := root.(neosynctypes.NeosyncMssqlValuer); ok {
+		value, err := valuer.ValueMssql()
+		if err != nil {
+			return nil, false, fmt.Errorf("unable to get MSSQL value from NeosyncMysqlValuer: %w", err)
```
Comment (nickzelei) [inline worker/pkg/benthos/sql/processor_neosync_mssql.go:153]:
```suggestion
			return nil, false, fmt.Errorf("unable to get MSSQL value from NeosyncMssqlValuer: %w", err)
```

File: worker/pkg/benthos/sql/processor_neosync_pgx.go [hunk 4]
```diff
@@ -145,28 +141,42 @@ func getPgxValue(value any, colDefaults *neosync_benthos.ColumnDefaultProperties
...
+	if value == nil {
+		return nil, nil
+	}
+
+	switch {
+	case pgutil.IsJsonPgDataType(datatype):
+		bits, err := json.Marshal(value)
 		if err != nil {
-			return nil, fmt.Errorf("unable to handle byte slice: %w", err)
+			return nil, fmt.Errorf("unable to marshal JSON: %w", err)
```
Comment (nickzelei) [inline worker/pkg/benthos/sql/processor_neosync_pgx.go:152]: like my comment above, this could be more descriptive.

File: backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go [hunk 5]
```diff
@@ -30,14 +32,21 @@ import (
...
 	myutil "github.com/nucleuscloud/neosync/internal/mysql"
 	pgutil "github.com/nucleuscloud/neosync/internal/postgres"
 	querybuilder "github.com/nucleuscloud/neosync/worker/pkg/query-builder"
+
 	"go.mongodb.org/mongo-driver/bson"
 	"golang.org/x/sync/errgroup"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+var _ = func() any { //nolint:unparam
```
Comment (nickzelei) [inline backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go:45]: this is strange.
how is this different than using `func init()` ?

File: internal/gob/gob.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	dynamotypes "github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/jackc/pgx/v5/pgtype"
+	neosynctypes "github.com/nucleuscloud/neosync/internal/neosync-types"
+)
+
+// need to register all the types that are used in the connection data service
+// because we use interfaces for the types
+func RegisterGobTypes() {
+	gob.Register(map[string]any{})
+	gob.Register(map[string]any{})
```
Comment (nickzelei) [inline internal/gob/gob.go:16]: why is this here twice?

File: internal/gob/gob.go [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	gob.Register(pgtype.Numeric{}) // TODO fix this
```
Comment (nickzelei) [inline internal/gob/gob.go:17]: what is wrong with this?

File: internal/gob/gob.go [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+	gob.Register([]any{})
+	gob.Register(time.Time{})
```
Comment (nickzelei) [inline internal/gob/gob.go:19]: interesting to see native go types being registered here. that is very strange that they arent natively supported by the language

**Review by nickzelei — COMMENTED** [review]
this turned into a bigger PR dinnit 😄 
left a few Qs, but otherwise it looks good.

File: internal/gob/gob.go (hunk 7 above)
Comment (alishakawaguchi) [reply internal/gob/gob.go:16]: in the connection data stream and neosync data stream endpoint we shouldn't have any connection specific types. It should all be go types but it seems this one slipped through. Was going to fix in a follow up pr. Might be related to the big int issue I've been seeing

File: internal/gob/gob.go (hunk 6 above)
Comment (alishakawaguchi) [reply internal/gob/gob.go]: oops

File: backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go (hunk 5 above)
Comment (alishakawaguchi) [reply backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go]: no this should use init. updated it

File: internal/gob/gob.go (hunk 7 above)
Comment (nickzelei) [reply internal/gob/gob.go:16]: sounds good
